### ISSUE TITLE
9.9 (LTS)/1.14.0 - Use of Reference Branch for New Code with SCM Disabled

### ISSUE THREAD

**Issue by nikleo-eng** [mc1arke/sonarqube-community-branch-plugin#1009]
**Describe the bug**
<ins>With SCM disabled</ins>, when I set for the new code of branches, eg child_branch, the reference to a master stable branch, eg parent_branch, the first analysis of child_branch shows correctly the new/modified files as new code with respect to parent_branch, but the second and successive analyses of child_branch not compare completely the all files of child_branch with the all files of parent_branch; in the second and successive analyses are showed as new code, not the new/modified files from the first anaylsis + the new/modified files (ever with respect to parent_branch) of these analyses, but only the new/modified files of these last analyses with respect to the respectively previous analysis.

So, if the quality gate of the first analysis is failed, the second analysis - that treats only the differences between the first and the second analysis - has a succeded quality gate (obviously if in the differences not are introduced issues that bring the new quality gate in failure); same behaviour between the second and the third analysis, and so on ... I think this behaviour is a bug.

**To Reproduce**
1. Create a project with new code setted on reference branch
2. Analyze the reference branch
3. Create a branch with some code added, with respect to the reference branch
4. Analyze the branch, the showed infos on the added code are correct
5. Add extra different new code to the branch (with respect to the reference branch and the code of 3.)
6. Analyze the branch again, the showed infos are relative only to the new code of 5. (and not to the new code of 3. + 5.)

**Expected behavior**
With the use of reference branch for new code, I'm expecting that every analysis of child_branch compare all files from child_branch with all files of parent_branch: i'm expecting that the analysis are ever full and not incremental with respect to previous analysis; in this way, the quality gate of every analysis shows the full situation of complete source code.

**Software Versions**
 - SonarQube Version: 9.9 (LTS)
 - Plugin Version: 1.14.0

**Comment by mc1arke** [comment]
I don't think this is something under the control of the plugin - the calculation of intersection/difference of issues is performed by Sonarqube core. Have you tried this in Sonarqube developer edition to confirm what it does?

**Comment by Polina-Pavan** [comment]
No.. Will update you once I try

**Comment by rwlrvns** [comment]
Reproduced in a dev edition. The main key here is the SCM integration: SQ uses these data to detect New Code. The [doc](https://docs.sonarsource.com/sonarqube-server/latest/analyzing-source-code/scm-integration/) states it uses the analysis dates when SCM data is not available, which explains why subsequent analyses do pass the QualityGate.

**Comment by mc1arke** [comment]
Sorry @rwalravens are you saying this works as you'd expect in the dev edition, or that the plugin and dev edition are doing the same thing?

**Comment by rwlrvns** [comment]
@mc1arke both. I tested this case in the dev edition, and it behaves the same as the community with plugin, which is as I'd expect from the documentation.

**Comment by mc1arke** [comment]
Thanks for the confirmation. Closing this since it's working as I would expect.
